django/contrib/admin/templates/admin/app_list.html

```diff
@@ -2,15 +2,15 @@
 
 {% if app_list %}
   {% for app in app_list %}
-    <div class="app-{{ app.app_label }} module{% if app.app_url in request.path %} current-app{% endif %}">
+    <div class="app-{{ app.app_label }} module{% if request and app.app_url in request.path %} current-app{% endif %}">
       <table>
         <caption>
           <a href="{{ app.app_url }}" class="section" title="{% blocktranslate with name=app.name %}Models in the {{ name }} application{% endblocktranslate %}">{{ app.name }}</a>
         </caption>
         {% for model in app.models %}
-          <tr class="model-{{ model.object_name|lower }}{% if model.admin_url in request.path %} current-model{% endif %}">
+          <tr class="model-{{ model.object_name|lower }}{% if request and model.admin_url and model.admin_url in request.path %} current-model{% endif %}">
```

Comment:
This isn't nice, especially not if we generalize over the entire set of old templates out there that are relying on the soft-fail here: it's old as the arc that you can define an if block that simple won't show if the variable is missing. 

I'm not sure that requiring `if var and var.attr` everywhere is a win. (In fact, stronger than that...)

Are there alternatives? I see that if `var.attr` raises (like an actual exception) we want that to raise, but can we maintain the historic/super behaviour of allowing `var` to not be defined?

Reply:
I see you're point that the quotes shouldn't be here. Thanks.

I changed this to check the variable is defined before use. I could also add a new test that shows that without the variable check, an exception is raised, just let me know.

If a variable is undefined, an `in` check shouldn't implicitly result in `False`. That could result in leaking private data. For example:

```
{% if thing.undefined in public_values %}
   <p>Public stuff</p>
{% else %}
   <p>Private stuff</p>
{% endif %}
```

The safest approach is raise an exception.

Reply:
These checks exist to facilitate testing d522b51c401429c169d88742178a9b3777903d9e. In hindsight, I think it would be simpler to always pass the `request` object to Django admin templates and drop the requirement on the context processor. Doing so would allow me to drop the `request` checks in this template. I've done this in the latest revision, which simplifies the changes to this template.